### PR TITLE
cmake: increase minimal required version to 3.20.0

### PR DESCRIPTION
**Pull request by tejlmand**
Move to CMake 3.20.0.

At the Toolchain WG it was decided to move to CMake 3.20.0.

The main reason for increasing CMake version is better toolchain
support.
    
Better toolchain support is added in the following CMake versions:
- armclang, CMake 3.15
- Intel oneAPI, CMake 3.20
- IAR, CMake 3.15 and 3.20
    

See MoM: https://docs.google.com/document/d/1IQKBK-GcJNZG0O9QArqYfvb6Huk5xHscN-XIGEZr-z8/edit#heading=h.gc6zh884razh

Note: ~DNM because the PR contains an unmerged manifest update.~
See: https://github.com/zephyrproject-rtos/lvgl/pull/13 <-- is merged and manifest is updated to SHA revision.

### CODE REVIEW

**Comment by tejlmand** [comment]
PR to update the CMake version in docker is found here: https://github.com/zephyrproject-rtos/docker-image/pull/71.

That should fix the CI issue regarding old CMake version in use.

**Review by gmarull — CHANGES_REQUESTED** [review]
nit: https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/getting_started/index.rst needs to be updated with new minimum version

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
:disappointed: 

I agree with @gmarull, but this is not just a nit.

The latest Ubuntu LTS is shipping an incompatible version:

```
$ apt-cache show cmake | grep Version
Version: 3.16.3-1ubuntu1
```

Zephyr v1.14 LTS documentation has a detailed HOWTO on how to get cmake from various different sources to get something more recent than the distribution version:

https://docs.zephyrproject.org/1.14.1/getting_started/installation_linux.html#install-requirements-and-dependencies

I would expect something similar to be resurrected here.

**Comment by galak** [comment]
@tejlmand you can use docker image v0.18.2 now.

**Comment by tejlmand** [comment]
> Zephyr v1.14 LTS documentation has a detailed HOWTO on how to get cmake from various different sources to get something more recent than the distribution version:
> 
> https://docs.zephyrproject.org/1.14.1/getting_started/installation_linux.html#install-requirements-and-dependencies
> 
> I would expect something similar to be resurrected here.

It already exists:
https://docs.zephyrproject.org/latest/getting_started/installation_linux.html#installation-linux

> A recent CMake version is required. Check what version you have by using cmake --version. If you have an older version, there are several ways of obtaining a more recent one:
>
> -  On Ubuntu, you can follow the instructions for adding the kitware third-party apt repository to get an updated version of cmake using apt.

But I will add more description on the getting started page.

**Comment by tejlmand** [comment]
> nit: https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/getting_started/index.rst needs to be updated with new minimum version

Done.

> But I will add more description on the getting started page.

and done.

**Comment by evgeniy-paltsev** [comment]
Just FYI: we use 3.20.2 cmake in our internal verification - so this change is OK for ARC.

File: doc/getting_started/index.rst [hunk 1]
```diff
@@ -76,6 +76,15 @@ The current minimum required version for the main dependencies are:
...
       .. _install_dependencies_ubuntu:
 
+      #. Download, inspect and execute the Kitware archive script to add the
+         Kitware APT repository to yoursources list::
+
+         .. code-block:: bash
+
+            wget https://apt.kitware.com/kitware-archive.sh
+	    # Inspect that script is safe and execute
+	    sudo bash kitware-archive.sh
```
[the remark this answers is not in the document]
Comment (tejlmand) [reply doc/getting_started/index.rst:86]: good point, no reason to mention it twice.

[the remark this answers is not in the document]
Comment (tejlmand) [reply doc/getting_started/index.rst]: done

File: doc/getting_started/index.rst [hunk 2]
```diff
@@ -95,6 +104,12 @@ The current minimum required version for the main dependencies are:
          Refer to the :ref:`installation_linux` page for additional information on updating
          the dependencies manually.
 
+	 .. note::
+
+	    It is also possible to execute each step in ``kitware-archive.sh``
+	    manually, see `kitware third-party apt repository <https://apt.kitware.com/>`_
+	    for detailed instructions.
+
```
[the remark this answers is not in the document]
Comment (tejlmand) [reply doc/getting_started/index.rst]: we could shorten the note to something like:
> A detailed explanation of ``kitware-archive.sh`` can be found here `kitware third-party apt repository` <https://apt.kitware.com/>

[the remark this answers is not in the document]
Comment (tejlmand) [reply doc/getting_started/index.rst]: Or even move it up, and remove the note, like:

> Download, inspect and execute the Kitware archive script to add the Kitware APT repository to your sources list.
> A detailed explanation of kitware-archive.sh can be found here [kitware third-party apt repository](https://apt.kitware.com/)
> ```
> wget https://apt.kitware.com/kitware-archive.sh
> sudo bash kitware-archive.sh
> ```

File: doc/getting_started/index.rst [hunk 3]
```diff
@@ -76,6 +76,15 @@ The current minimum required version for the main dependencies are:
 
       .. _install_dependencies_ubuntu:
 
+      #. Download, inspect and execute the Kitware archive script to add the
+         Kitware APT repository to yoursources list::
```
[the remark this answers is not in the document]
Comment (tejlmand) [reply doc/getting_started/index.rst]: done

File: doc/getting_started/index.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (gmarull) [reply doc/getting_started/index.rst]: Yep, sounds good:

> Or even move it up, and remove the note, like:
> 
> > Download, inspect and execute the Kitware archive script to add the Kitware APT repository to your sources list.
> > A detailed explanation of kitware-archive.sh can be found here [kitware third-party apt repository](https://apt.kitware.com/)
> > ```
> > wget https://apt.kitware.com/kitware-archive.sh
> > sudo bash kitware-archive.sh
> > ```

[the remark this answers is not in the document]
Comment (tejlmand) [reply doc/getting_started/index.rst]: done

File: doc/getting_started/index.rst [hunk 4]
```diff
@@ -76,6 +76,16 @@ The current minimum required version for the main dependencies are:
 
       .. _install_dependencies_ubuntu:
 
+      #. Download, inspect and execute the Kitware archive script to add the
+         Kitware APT repository to your sources list::
```
[the remark this answers is not in the document]
Comment (tejlmand) [reply doc/getting_started/index.rst:80]: I was unsure :man_shrugging: so I just kept them.
Do you prefer them to be removed completely, or moved to the line below ?

[the remark this answers is not in the document]
Comment (gmarull) [reply doc/getting_started/index.rst:80]: `::` are used to introduce literals, see https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks

[the remark this answers is not in the document]
Comment (tejlmand) [reply doc/getting_started/index.rst:80]: I had the impression it was simply to get a single `:` at the end, but I see that is also the result.
> The handling of the :: marker is smart:
> - ...
> - If it is preceded by non-whitespace, the marker is replaced by a single colon.

Have moved in to line below.

**Review by gmarull — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by tejlmand** [comment]
Buildkite failure is due to timeouts because of the amount of files changed --> CI runs for long time.

**Review by nashif — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
@tejlmand consider sending email when this is merged so people know they need to update.

**Pull request merged**